contents/blog/mobile-session-replay.md

```diff
@@ -0,0 +1,139 @@
+
+There are even breaking differences within platforms. For example, Jetpack Compose uses a compositional model for UI, which is different from Android's traditional view-based model. This means you need to develop separate ways of doing replays when using it. iOS has a similar problem with SwiftUI versus UIKit.
+
+### 2. Performance
+
+Phones are much less powerful than desktops. Because of this, we need to be much more sensitive about performance. 
+
+If you've ever tried to record your phone screen, you would know its impact on performance. Apps take longer to load, animations become choppy, reactivity degrades, and your phone heats up.
+
+Anything that degrades your user experience is not an option for many developers.
+
+### 3. Privacy
+
+To protect users privacy, developers often mask and exclude sensitive information. The tricky part of this is identifying what information is sensitive.
+
+Web apps rely on the DOM to do this. The DOM provides a hierarchical tree structure that represents and uniquely identifies elements. The web also has standardized elements, like `<input type="password">`. Combining these makes it easier to identify, mask, and exclude sensitive elements on the web.
+
+Mobile doesn't have standardized structures or elements. Accessibility identifiers are also inconsistently implemented. This means identifying, masking, and excluding content is a lot trickier.
+
+### 4. Testing
+
+We're big fans of [dogfooding](/product-engineers/dogfooding) at PostHog. Often, we are our own best users, but we don't have a mobile app.
+
+This meant that during development we relied on demo and open source apps. This risks creating something that doesn't work well for larger, production-quality apps.
+
+Developing a high-quality mobile replay product means relying more on our users and their feedback.
+
+## How we solved these problems
+
+### 1. Multiple platforms
+
+The solution to this problem is plastered on motivation posters everywhere: **do the work** (to develop for each platform).
+
+The one thing that made this easier was reusing our existing session replay infrastructure. This included the data structure, the way to store replays, as well as a complete product for playback and analysis.
+
+![Replay](https://res.cloudinary.com/dmukukwp6/image/upload/replay_03a8c56981.png)
+
+The first platform-specific work we did was rewriting the SDKs in Kotlin and Swift, removing code we didn't use, improving tests, automating deployments, and making sure they worked with the latest platforms.
+
+Once updated, we could get on with the work developing platform-specific proof-of-concepts.
+
+### 2. Performance
+
+Not slowing down people's apps was core to our mission with session replay. 
+
+To do this, we decided to focus on capturing wireframes instead of screenshots. Although we built both in the end, wireframe mode is much less performance-intensive, meaning it is the mode users prefer.
```

Comment:
can we quantify this? how big is the difference?

Reply:
Checking: https://posthog.slack.com/archives/C03PB072FMJ/p1728422422040949